src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3RequestStream.cs

```diff
@@ -1064,6 +1064,7 @@ private int ReadResponseContent(HttpResponseMessage response, Span<byte> buffer)
                     if (_responseDataPayloadRemaining <= 0 && !ReadNextDataFrameAsync(response, CancellationToken.None).AsTask().GetAwaiter().GetResult())
                     {
                         // End of stream.
+                        _connection.RemoveStream(_stream);
```

Comment:
I am not 100% sure if I need to care about duplex Http content here (and therefore check if the client-to-server direction of the stream has been completed as well.). The tests seem to run fine atm.

Reply:
Based on this in H/2 (if I'm interpreting it correctly):
https://github.com/dotnet/runtime/blob/62eb291312749c52507309d50051dd61538cc62d/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs#L372-L380

Both sides need to be completed before the request gets removed